lib/angular/fieldLocation/fieldLocationController.js

```diff
@@ -15,19 +15,43 @@ function FieldLocationController($scope) {
   fieldLocationCtrl.setLocation = function(event) {
     event.stopPropagation();
 
-    //TODO Move it to a service to get the location
-    navigator.geolocation.getCurrentPosition(function(pos) {
+    /*
+    * Get the current position from the Geolocation Service.
+    */
+    GeolocationService.getGeolocation(function(err, response) {
+      if (response) {
+        console.log('Position Obtained');
+        fieldLocationCtrl.applyValues(response.coords.latitude, response.coords.longitude);
+      } else {
+        console.log(err.message ? err.message : err);
+        // location not available, set a default
+        fieldLocationCtrl.applyValues(-1, -1);
```

Comment:
If there is an error from the geolocation service, it should display an error on the field that the user can read.

This will show them more feedback.